src-cpp/rdkafkacpp.h

```diff
@@ -2931,6 +2931,31 @@ class RD_EXPORT KafkaConsumer : public virtual Handle {
    */
   virtual Error *incremental_unassign(
       const std::vector<TopicPartition *> &partitions) = 0;
+
+  /**
+   * @brief Close and shut down the consumer.
+   *
+   * Same behaviour as RdKafka::KafkaConsumer::close() but with the following
+   * changes:
+   *
+   * Rebalance events/callbacks (etc) will be forwarded to the
+   * application-provided \p queue. The application must poll this queue (in
+   * another thread) until this close function returns or
```

Comment:
> in another thread

two thoughts:

it would be better UX if close(queue) was non-blocking, then you wouldn't need another thread.

they way you describe it, it seems like you need to check two conditions in order to decide when to exit the poll loop.